gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDataset.java

```diff
@@ -151,4 +172,105 @@ public static String resolveTemplate(String rawString, Table table) {
     }
     return StringUtils.replaceEach(rawString, new String[] { DATABASE_TOKEN, TABLE_TOKEN }, new String[] { table.getDbName(), table.getTableName() });
   }
+
+  /***
+   * Parse logical Database and Table name from a given DbAndTable object.
+   *
+   * Eg.
+   * Dataset Name Pattern         : prod_$LOGICAL_DB_linkedin.prod_$LOGICAL_TABLE_linkedin
+   * Source DB and Table          : prod_dbName_linkedin.prod_tableName_linkedin
+   * Logical DB Token             : $LOGICAL_DB
+   * Logical Table Token          : $LOGICAL_TABLE
+   * Parsed Logical DB and Table  : dbName.tableName
+   *
+   * @param datasetNamePattern    Dataset name pattern.
+   * @param dbAndTable            Source DB and Table.
+   * @param logicalDbToken        Logical DB token.
+   * @param logicalTableToken     Logical Table token.
+   * @return  Parsed logical DB and Table.
+   */
+  @VisibleForTesting
+  public static DbAndTable parseLogicalDbAndTable(String datasetNamePattern, DbAndTable dbAndTable,
+      String logicalDbToken, String logicalTableToken) {
+    Preconditions.checkArgument(StringUtils.isNotBlank(datasetNamePattern), "Dataset name pattern must not be empty.");
+
+    List<String> datasetNameSplit = Lists.newArrayList(SPLIT_ON_DOT.split(datasetNamePattern));
+    Preconditions.checkArgument(datasetNameSplit.size() == 2, "Dataset name pattern must of the format: "
```

Comment:
You can have LoginEvent_prod and $LOGICAL_TABLE_prod pattern. 
Splitting by token will always yield two parts (because notice we do not emit empty strings), so here the splits will be prefix: "" and postfix: "_prod" which will be removed from LoginEvent_prod yielding LoginEvent. There is a test to cover this as well.